apa102/src/main/java/com/google/android/things/contrib/driver/apa102/Apa102.java

```diff
@@ -212,9 +242,9 @@ public void write(int[] colors) throws IOException {
         pos += APA_START_FRAME_PACKET_LENGTH;
 
         // Compute the packets to send.
-        byte brightness = (byte) (0xE0 | mLedBrightness); // Less brightness possible
         final Direction currentDirection = mDirection; // Avoids reading changes of mDirection during loop
         for (int i = 0; i < colors.length; i++) {
+            byte brightness = (byte) (0xE0 | mLedBrightness[i]); // Less brightness possible
```

Comment:
in this case I think an Index out of bounds exception is fine (after the changes). The developer is passing two arrays, one for brightness and one for colors, both should be of the same length.

Reply:
I'm not sure throwing an exception is the best option. This method previously could never throw that exception, so ideally we wouldn't change that behavior. (Even if we do decide to throw, we should be the ones throwing it so that we can provide an appropriate message.)

One alternative, for example, would be to repeat the final brightness value.

Reply:
I'm more in favor of returning an exception of the length of the arrays is different, so it does not behave unexpectedly, will implement that later.

Reply:
so, @jdkoren I think this is good now, right?